### ISSUE TITLE
Add task status filters to Task Duration Bar chart view 

### ISSUE THREAD

**Issue by ketozhang** [apache/airflow#40445]
### Description

Currently the Task Duration bar chart shows every task instance. The only filter available are DAG type and state, but there is no way to filter task-level metadata.

### Use case/motivation

For example, adding a task status filter, we can filter out any task that were skipped.

Here is what it currently looks like with a DAG that skip this task more often than not...

![image](https://github.com/apache/airflow/assets/13506487/e262c47e-ba43-46ff-a915-f0fb81369791)
You must agree the histogram is not so useful with all these skipped tasks.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by tirkarthi** [comment]
Currently the chart just takes the task instance row and plot the duration. Maybe something like filters for the overall grid will make this easier for the plot.

Related : https://github.com/apache/airflow/issues/38032

@bbovenzi

**Comment by bbovenzi** [comment]
I agree. The grid view is the main interface to navigate around a DAG. Let's add filters to that and have the task duration page use those filters.

As I wrote [here](https://github.com/apache/airflow/pull/40755#issuecomment-2228659336), we could also combined the task and run duration pages and included chart-specific filters too

**Comment by bbovenzi** [comment]
Closing. This is Airflow 2. Our charts are different in Airflow 3, and can be improved but in some different ways.
